expenseTracker/app.js

```diff
@@ -0,0 +1,41 @@
+require('dotenv').config()
+require('express-async-errors')
+
+const express = require('express')
+const app = express()
+
+const connectDB = require('./db/connect')
+const notFoundMiddleware = require('./middleware/not-found')
+const errorHandlerMiddleware =require('./middleware/error-handler')
+
+//routes
+const expenseRoutes = require('./routes/expenses')
+const authRoutes = require('./routes/authRoutes')
+
+//middleware
+app.use(express.static('./public'))
+
+//middleware to parse JSON
+app.use(express.json())
+
+//base routes
+app.use('/api/v1/expenses', expenseRoutes)
```

Comment:
You could just pass your auth middleware here on the entirety of the expenses route.